examples/data-objects/clicker/src/index.tsx

```diff
@@ -23,19 +26,29 @@ export class Clicker extends DataObject implements IFluidHTMLView {
     public get IFluidHTMLView() { return this; }
 
     private _counter: SharedCounter | undefined;
+    private _taskManager: TaskManager | undefined;
 
     /**
      * Do setup work here
      */
     protected async initializingFirstTime() {
         const counter = SharedCounter.create(this.runtime);
         this.root.set(counterKey, counter.handle);
+        const taskManager = TaskManager.create(this.runtime);
+        this.root.set(taskManagerKey, taskManager.handle);
     }
 
     protected async hasInitialized() {
         const counterHandle = this.root.get<IFluidHandle<SharedCounter>>(counterKey);
         this._counter = await counterHandle?.get();
-        this.setupAgent();
+        const taskManagerHandle = this.root.get<IFluidHandle<TaskManager>>(taskManagerKey);
+        this._taskManager = await taskManagerHandle?.get();
```

Comment:
Does this work? I do not remember when hasInitialized() is called, but I run into the issues (maybe earlier in init sequence) where it would deadlock, as resolution of handles relies on data store runtime to be fully initialized (and "visible" from container runtime) for handles in it to be resolved, as all handles have absolute path and depend on ability of container runtime to "see" fully initialized data store to resolve any handles.

Reply:
I haven't seen any deadlocks in my experience -- we follow this pattern in a number of our examples so if there's an issue then we probably need to revise this flow broadly.